source/common/common/utility.h

```diff
@@ -632,4 +633,64 @@ template <class Value> struct TrieLookupTable {
   TrieEntry<Value> root_;
 };
 
+// Helper class for allocating classes with variable-sized inlined storage. See
+// InlineString below for usage.
+class InlineStorage : public NonCopyable {
```

Comment:
Though there is InlineString as an example of using this class, I think it could be better if you explicitly point out that the derived class should 
1. hide its constructor and have a public interface like c`create(size_t buffer_size, ...)`
2. the buffer member has to be the last field of the class.
3. take care of buffer alignment.
It would be great if we can enforce this in the code, but I couldn't think of a way to achieve that.

Reply:
Good point, added this class doc:

```
// Mix-in class for allocating classes with variable-sized inlined storage.
//
// Use this class by inheriting from it, ensuring that:
//  - The variable sized array is declared as VarType[] as the last
//    member variable of the class.
//  - YourType accurately describes the type that will be stored there,
//    to enable the compiler to perform correct alignment. No casting
//    should be needed.
//  - The class constructor is private, because you need to allocate the
//    class the placed new operator exposed in the protected section below.
//    Constructing the class directly will not provide space for the
//    variable-size data.
//  - You expose a public factory method that return a placement-new, e.g.
//      static YourClass* alloc(size_t num_elements, constructor_args...) {
//        new (num_elements * sizeof(VarType)) YourClass(constructor_args...);
//      }
//
// See InlineString below for an example usage.
```